tests/sentry/api/endpoints/test_organization_details.py

```diff
@@ -798,11 +799,69 @@ def test_update_name_with_mapping(self):
         org = Organization.objects.get(id=organization_id)
         assert org.name == "SaNtRy"
 
+        # Process outbox to generate organization mapping
+        with outbox_runner():
+            pass
+
         with exempt_from_silo_limits():
             assert OrganizationMapping.objects.filter(
                 organization_id=organization_id, name="SaNtRy"
             ).exists()
 
+    def test_update_status_with_mapping(self):
+        org = Organization.objects.get(id=self.organization.id)
+        org.status = OrganizationStatus.PENDING_DELETION
+        org.save()
+        response = self.get_success_response(self.organization.slug, name="SaNtRy")
+
+        organization_id = response.data["id"]
+
+        org.refresh_from_db()
+        assert org.name == "SaNtRy"
+
+        # Process outbox to generate organization mapping
+        with outbox_runner():
+            pass
+
+        with exempt_from_silo_limits():
+            org_mapping_query = OrganizationMapping.objects.filter(
+                organization_id=organization_id, name="SaNtRy"
+            )
+            assert org_mapping_query.exists()
+            assert len(org_mapping_query) == 1
+            org_mapping = org_mapping_query.first()
+            assert org_mapping.status == OrganizationStatus.PENDING_DELETION
+
+    def test_slug_update_with_idempotency_key_matching_different_validated_org_mapping(self):
+        org_to_update = Organization.objects.get(id=self.organization.id)
+        existing_slug_name = org_to_update.slug
+        existing_org_name = org_to_update.name
+
+        OrganizationMapping.objects.create(
+            organization_id=999,
+            slug="taken-santry",
+            verified=True,
+            region_name="us",
+            idempotency_key="bad-key",
+        )
```

Comment:
When mappings are verified by celery tasks their idempotency key values are cleared. When would a verified record also have an idempotency key?

Reply:
I didn't realize that was the behavior and just assumed we never cleared the original key. I'll remove this test if it isn't a valid case

Reply:
I don't think this scenario can happen because of how the [verification tasks work](https://github.com/getsentry/sentry/blob/8f7f06dae432790d73a2e971b9933f8b9ca342ca/src/sentry/tasks/organization_mapping.py#L33-L45).